### PR TITLE
Feature/33856 Generated md5 assets hash and use it for cache busting

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Generate md5 hash from both js and css contents
- Use the hash for cache busting

### CODE REVIEW

File: admin/package.json [hunk 1]
```diff
@@ -150,5 +156,6 @@
       "<rootDir>/services/data-access"
     ],
     "verbose": false
-  }
-}
+  },
+  "mtc[assets-version]": "2c8bde8dabf32d6b9b64059875461c22"
```
Comment (GuyHarwood) [inline admin/package.json]: why the duplication?

Comment (GeorgeChatzigiannis) [reply admin/package.json]: thanks for spotting this, it was a leftover from a test I made.
it's removed now.

File: admin/gulpfile.js [hunk 2]
```diff
@@ -154,12 +157,37 @@ gulp.task('realclean', gulp.series('clean'), function () {
...
+  dir.readFiles('./public/', {
+    // match only filenames with a .js and .css extensions and that don't start with a `.´
+    match: /\.(js|css)(?:\?.*|)$/i,
+    exclude: /^\./
+  }, function (err, content, next) {
+    if (err) throw err
+    assetBuffer = assetBuffer + content
+    next()
+  },
+  function (err, files) {
```
Comment (jon-shipley) [inline admin/gulpfile.js]: `files` doesn't appear to be used.

File: admin/gulpfile.js [hunk 3]
```diff
@@ -154,12 +157,37 @@ gulp.task('realclean', gulp.series('clean'), function () {
     .pipe(clean())
 })
 
+gulp.task('generate-assets-version', function (done) {
+  let assetBuffer = ''
+  dir.readFiles('./public/', {
+    // match only filenames with a .js and .css extensions and that don't start with a `.´
+    match: /\.(js|css)(?:\?.*|)$/i,
```
[the remark this answers is not in the document]
Comment (GeorgeChatzigiannis) [reply admin/gulpfile.js]: @jon-shipley 🏅

File: admin/gulpfile.js [hunk 4]
```diff
@@ -154,12 +157,37 @@ gulp.task('realclean', gulp.series('clean'), function () {
...
+    exclude: /^\./
+  }, function (err, content, next) {
+    if (err) throw err
+    assetBuffer = assetBuffer + content
```
[the remark this answers is not in the document]
Comment (GeorgeChatzigiannis) [reply admin/gulpfile.js]: I forgot to rename the var as it is no longer a buffer while using this library. Thanks!

**Pull request merged**